### PR TITLE
Rs/wx/rule21 6

### PR DESCRIPTION
**Pull request by weilixu**

### CODE REVIEW

**Review by yunjoonjung-PNNL — APPROVED** [review]
I looked at the ```section21rule6.py``` file carefully and it looks perfect to me. Thanks for this effort Weili.

File: rct229/rules/section21/section21rule6.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+        def rule_check(self, context, calc_vals=None, data=None):
+            boiler_1_operation_lower_limit = calc_vals["boiler_1_operation_lower_limit"]
+            boiler_1_operation_upper_limit = calc_vals["boiler_1_operation_upper_limit"]
+            boiler_1_rated_capacity = calc_vals["boiler_1_rated_capacity"]
+            boiler_2_operation_lower_limit = calc_vals["boiler_2_operation_lower_limit"]
+            boiler_2_operation_upper_limit = calc_vals["boiler_2_operation_upper_limit"]
+            boiler_2_rated_capacity = calc_vals["boiler_2_rated_capacity"]
+            return (
+                boiler_1_operation_lower_limit == 0
```
Comment (charliepnnl) [inline rct229/rules/section21/section21rule6.py:111]: 0 should be ZERO.POWER here and below.

File: rct229/rules/section21/section21rule6.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+                and boiler_1_operation_upper_limit == boiler_1_rated_capacity
+                and boiler_2_operation_lower_limit == boiler_2_rated_capacity
```
Comment (charliepnnl) [inline rct229/rules/section21/section21rule6.py:113]: I question whether the logic is correct in the RDS.  Unless I am not understanding, which is likely, shouldn't the logic be:
```
and boiler_2_operation_lower_limit == boiler_1_rated_capacity
and boiler_2_operation_upper_limit == boiler_1_rated_capacity + boiler_2_rated_capacity
```
?

**Review by charliepnnl — APPROVED** [review]
Approved with two comments that require attention.

File: rct229/rules/section21/section21rule6.py (hunk 2 above)
Comment (weilixu) [reply rct229/rules/section21/section21rule6.py:113]: I do have the same question, I am reserving this PR for this Thursday RDS meeting.

Comment (weilixu) [reply rct229/rules/section21/section21rule6.py:113]: change to 
`
            - Case 1: If boiler_1's operation lower limit is 0 and operation upper limit is equal to its rated capacity, and boiler_2's operation lower limit is equal to boiler_1 rated capacity and operation upper limit is equal to the sum of boiler 1 and boiler 2 rated capacity: `if ( boiler_1.operation_lower_limit == 0 ) AND ( boiler_1.operation_upper_limit == boiler_1.rated_capacity ) AND ( boiler_2.operation_lower_limit == boiler_1.rated_capacity ) AND ( boiler_2.operation_upper_limit == boiler_21.rated_capacity + boiler_2.rated_capacity ): PASS`

            - Case 2: Else if boiler_2's operation lower limit is 0 and operation upper limit is equal to its rated capacity, and boiler_1's operation lower limit is equal to boiler_2 rated capacity and operation upper limit is equal to the sum of boiler 1 and boiler 2 rated capacity: `if ( boiler_2.operation_lower_limit == 0 ) AND ( boiler_2.operation_upper_limit == boiler_2.rated_capacity ) AND ( boiler_1.operation_lower_limit == boiler_2.rated_capacity ) AND ( boiler_1.operation_upper_limit == boiler_1.rated_capacity + boiler_2.rated_capacity ): PASS`
`

**Pull request merged**